docs/guides/rendering.md

```diff
@@ -1,445 +1,350 @@
-
-One caching issue is figuring out how to combine the `ChunkCache` with
-Dasks's built-in caching. We probably want to keep the `ChunkCache` for
-rendering non-Dask arrays? But when using Dask, we defer to its cache? We
-certainly don't want to cache the data in both places.
-
-Another issue is whether to cache `OctreeChunks` or tiles in the visual,
-beyond just caching the raw data. If re-creating both is fast enough, the
-simpler thing is evict them fully when a chunk falls out of view. And
-re-create them if it comes back in view. It's simplest to keep nothing but
-what we are currently drawing.
-
-However if that's not fast enough, we could have a MRU cache of
-`OctreeChunks` and tiles in VRAM, so that reviewing the same data is
-nearly instant. This is adding complexity, but the performance might be
-worth it.
+These data coordinates are still continuous values that may not perfectly align with data array indices and may even fall outside of the valid range of the layer's data array.
+But after clamping and rounding the coordinates, the resulting indices can be used to look-up the sub-set of the layer's data that are in view.
+The exact form of this look-up depends on if the layer is an image-like layer.
+
+When a layer's transform state includes non-trivial rotations, slicing is limited as described in some related issues (e.g. [#2616](https://github.com/napari/napari/issues/2616)).
+That's because the slicing operation is no longer selecting an axis-aligned region of the layer's data.
+While there are some ideas to improve this (see [#3783](https://github.com/napari/napari/issues/3783)), there are no active efforts in development.
+
+### Loading layer data
+
+Once we have the slice indices into a layer's data, we need to load the corresponding region of data into RAM.
+
+### Loading array-like image data
+
+{class}`Image<napari.layers.Image>` layer data does not have a single specific type (e.g. numpy's `ndarray`).
+Instead it must only have the attributes and methods defined in [`LayerDataProtocol`](https://github.com/napari/napari/blob/eab7661459e70479c7c7d587a36463f3b099b64a/napari/layers/_data_protocols.py#L51).
+
+Numpy's `ndarray` is compatible with this protocol, but so are array types from other packages like [Dask](https://docs.dask.org/en/latest/array.html), [Zarr](https://zarr.readthedocs.io/en/stable/_autoapi/zarr.core.Array.html), and more.
+This flexibility allows you refer to image data that does not fit in memory or still needs to be lazily computed, without complicating napari's core implementation at all.
+
+However, it also means that simply reading image data may be slow because the data must be read from disk, downloaded across a network, or calculated from a compute graph.
+This means array accesses can take an arbitrary long time to complete.
+
+### Loading multi-scale image data
+
+`Image` layers also support multi-scale image data, where multiple resolutions of the same image content are stored.
```

Comment:
True also for labels right? Maybe worth mentioning once somewhere?

Reply:
Yes, excellent points. With the caveat that one can't paint into multiscale labels.

Reply:
I added `Labels` here to make this clear. There are a lot of details in this document already, so I did not add the caveat because there are many many more.